core/server/auth/oauth.js

```diff
@@ -65,12 +66,12 @@ function exchangePassword(client, username, password, scope, body, authInfo, don
 }
 
 function exchangeAuthorizationCode(req, res, next) {
+    var ip = req.connection.remoteAddress || req.ip;
```

Comment:
in the brute middleware we set `req.authInfo.ip`.
can't we use this here as well?